### PR TITLE
Created service package

### PR DESCRIPTION
**Pull request by 0xKrishna**
# Description

Restructuting changes to heimdalld folder, So that we can start heimdall process from other services/scripts.
- `NewHeimdallService` as a constructor method to start the process
- `GetHeimdallApp` to get the heimdall app object in external services

We have also removed code dependency on bor repo, Moved all the required structs which we were using from bor repo to `bor/types` in heimdall repo itself.

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have added at least 2 reviewer or the whole pos-v1 team
- [ ] I have added sufficient documentation in code
- [x] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

# Testing

- [ ] I have added unit tests
- [ ] I have added tests to CI
- [x] I have tested this code manually on local environment
- [ ] I have tested this code manually on remote devnet using express-cli
- [x] I have tested this code manually on mumbai
- [ ] I have created new e2e tests into express-cli

### CODE REVIEW

File: bor/beginblocker.go [hunk 1]
```diff
@@ -22,7 +21,7 @@ func BeginBlocker(ctx sdk.Context, _ abci.RequestBeginBlock, k Keeper) {
 			return
 		}
 
-		var spans []*bor.ResponseWithHeight
+		var spans []*types.ResponseWithHeight
```
Comment (JekaMas) [inline bor/beginblocker.go:24]: why don't keep it in Bor package or, if needed, bor/api ?

**Review by JekaMas — APPROVED** [review]

File: bor/beginblocker.go (hunk 1 above)
Comment (0xKrishna) [reply bor/beginblocker.go:24]: That creates dependency issue, I tried.
Then moved it to heimdall repo and it worked.

File: bor/client/rest/query.go [hunk 2]
```diff
@@ -534,13 +532,13 @@ func loadSpanOverrides() {
 		return
 	}
 
-	var spans []*bor.ResponseWithHeight
+	var spans []*types.ResponseWithHeight
```
[the remark this answers is not in the document]
Comment (0xKrishna) [reply bor/client/rest/query.go:535]: Okay

File: bor/beginblocker.go (hunk 1 above)
Comment (JekaMas) [reply bor/beginblocker.go:24]: Even a new package like bor/api?

Comment (0xKrishna) [reply bor/beginblocker.go:24]: Okay, Let me try that

Comment (0xKrishna) [reply bor/beginblocker.go:24]: While I was checking this I noticed that in latest bor code (0.3.x) we have removed `ResponseWithHeight` from the code, There is no struct like that anymore. I am looking more analyse the impact of that on heimdall code.

Comment (0xKrishna) [reply bor/beginblocker.go:24]: So in latest version of bors we don't have these structs which I have replaced, So I think we will anyway have to update these after we release v0.3.0 on bor and update that in heimdall. I will create a task for that but for now I think the change I did should be fine.

**Pull request merged**